.github/actions/install-python-and-package/action.yml

```diff
@@ -16,20 +16,32 @@ runs:
       uses: actions/setup-python@v4
       with:
         python-version: ${{ inputs.python-version }}
-    
+
     - uses: actions/cache@v3
       with:
         path: ${{ env.pythonLocation }}
         key: ${{ env.pythonLocation }}-${{ hashFiles('setup.cfg') }}
-    
+
     - name: Python info
       shell: bash {0}
       run: |
         which python3
         python3 --version
-    
-    - name: Upgrade pip and install dependencies
+
+    - name: Upgrade pip
       shell: bash {0}
       run: |
         python3 -m pip install --upgrade pip setuptools wheel
+
+    # only necessary on linux to avoid bloated installs
+    - name: Install tensorflow/pytorch cpu version
```

Comment:
Great that we keep running gpu versions on Windows/Mac. This way our code is run with both cpu and gpu versions. Of course this can complicate things when certain builds start failing while others pass, but there's always something. Overall, I like it!

Reply:
I don't know where it actually decides to run on the cpu or gpu. The VMs that the workflows run on do not have access to GPUs as far as I'm aware, so I think we are running cpu code only.